docs/current_docs/manuals/developer/ide-integration.mdx

```diff
@@ -43,69 +43,151 @@ echo go.work.sum >> .gitignore
 You can also choose to omit the local Go files generated by `dagger develop` from your version control system, as they aren't required to run the module.
 :::
 </TabItem>
+
 <TabItem value="Python">
-To get your IDE to recognize a Dagger Python module and any added Dagger module dependencies in the generated client, follow the steps below for your package manager.
+To get your IDE to recognize a Dagger Python module and any added Dagger module dependencies in the generated client, all dependencies must be installed in an **activated** [virtual environment](https://packaging.python.org/en/latest/tutorials/installing-packages/#creating-virtual-environments) (for example, in `.venv`, next to `pyproject.toml`). This can either be done manually or transparently by a package manager.
+
+For example, to open the Visual Studio Code from the terminal, with functioning autocompletions (assuming the [Python extensions](https://github.com/microsoft/vscode-python) are installed):
+
+```shell
+dagger develop
+uv run code .
+```
+
+For NeoVim, just replace `code` with `vim`, or any other terminal based editor.
+
+:::note
+The `uv run` command works with a `uv.lock` file. If you need to use a `requirements.lock` or other, continue reading for more examples.
+:::
+
+## Project environment
+
+The following examples show how to get a working [project environment](https://docs.astral.sh/uv/concepts/projects/#project-environments) (`.venv`) from the terminal, just make sure you run `dagger develop` first, to get the necessary `sdk` directory locally.
 
 <Tabs groupId="pytools">
-<TabItem value="pip">
+<TabItem value="uv">
 
-Install the Python SDK in a [virtual environment](https://packaging.python.org/en/latest/tutorials/installing-packages/#creating-virtual-environments) from the generated `./sdk` directory, as shown below:
+When using a `uv.lock`:
 
 ```shell
-python -m venv .venv
-source .venv/bin/activate
-pip install -e ./sdk
+uv sync
 ```
+
+If you have an older Dagger module, without a `uv.lock` file, you can create it with:
+
+```shell
+uv add --editable ./sdk
+rm requirements.lock
+````
+
+:::note
+Uv with `uv.lock` needs to have the SDK library (`dagger-io`, generated in `./sdk`) defined as a **production** dependency, unlike the other installation methods where it should be a development dependency.
+:::
+
 </TabItem>
-<TabItem value="Poetry">
+<TabItem value="uv pip">
+
+When using a `requirements.lock` file, uv provides a faster ["pip interface"](https://docs.astral.sh/uv/pip/), that replaces multiple tools. It can manage the Python installation, virtual environment creation, and dependency installation.
```

Comment:
```suggestion
When using a `requirements.lock` file, uv provides a faster ["pip interface"](https://docs.astral.sh/uv/pip/) that replaces multiple tools. It can manage the Python installation, virtual environment creation, and dependency installation.
```